cpp/include/IceUtil/Timer.h

```diff
@@ -12,6 +12,7 @@
 
 #include <set>
 #include <map>
+#include <functional>
```

Comment:
This missing header was causing build failures with recent Visual Studio 2023 versions.

The vcpkg already has this fix https://github.com/microsoft/vcpkg/commit/fbaaad1eb1d7ebc576bcc842aa1bac1c9e76239e